src/viewer.js

```diff
@@ -37,3 +37,28 @@ if (OCA.Viewer) {
 		theme: 'light',
 	})
 }
+
+// TODO: Viewer.openWith introduced with https://github.com/nextcloud/viewer/pull/1273
+//       This check can be replaced with `if(OCA.Viewer)` once NC 24 is EOL.
+if (OCA.Viewer.openWith) {
+	const supportedMimes = OC.getCapabilities().richdocuments.mimetypes.concat(OC.getCapabilities().richdocuments.mimetypesNoDefaultOpen)
```

Comment:
We should only register optional ones here. Additionally we should probably add pdf to the default mime types if files_pdfviewer is disabled.

Reply:
In the case onlyoffice and collabora are both installed I think onlyoffice usually is set as the default for opening Office Open XML mimetypes.

If we only register the optional ones, then there would be no way to open docx with collabora in this case.

Reply:
I'd say this is something we should then address separately. Maybe we can indeed have a setting for that or detect if there is an other app in use (ref #2301)

Reply:
My main concern is that we add the extra action also in cases when it is the only option to open with Collabora (which is probably the most common case).

Reply:
Makes sense, fixed!